src/bp/core/services/kvs.ts

```diff
@@ -28,22 +28,24 @@ export class KeyValueStore {
       keyCol: 'key',
       valueCol: 'value',
       modifiedOnCol: 'modified_on',
+      idCol: 'id',
       botId,
       key,
       value: safeStringify(value),
-      now: this.database.knex.date.now()
+      now: this.database.knex.date.now(),
+      id: `${botId}:${key}`
```

Comment:
It would be better practice to create a composite primary key on kvs instead. Just like you did with the Id, the primary key would be composed of the `botId` and the `key`.